Common++/src/IpAddress.cpp

```diff
@@ -364,15 +367,15 @@ namespace pcpp
 	{
 		memset(m_Mask, 0, IPV6_ADDR_SIZE);
 		int remainingPrefixLen = prefixLen;
-		for (auto byteIndex = 0; byteIndex < IPV6_ADDR_SIZE; byteIndex++)
+		for (unsigned char& byteIndex : m_Mask)
```

Comment:
We can rename `byteIndex` to `byte` and use `auto&`:
```cpp
for (auto& byte : m_Mask)
{
    ...
}
```

Reply:
fixed with 3ec344e61e6259128576e06d809e58ae62a9cc0b